google-cloud-bigquery/lib/google/cloud/bigquery/job.rb

```diff
@@ -239,12 +244,16 @@ def reload!
         #                               format: "json"
         #   extract_job.wait_until_done!
         #   extract_job.done? #=> true
-        def wait_until_done!
-          backoff = ->(retries) { sleep 2 * retries + 5 }
+        def wait_until_done! min: 5, max: Float::INFINITY
+          backoff = ->(retries) { 2 * retries + min }
```

Comment:
Since we are allowing these values to be set, we still have the magical number `2` which is the multiplier. I think this should also be exposed, so users can determine how they want the delay to increase.